scripts/components/color-picker-component/color-picker-component.js

```diff
@@ -0,0 +1,96 @@
+import React, { useRef } from 'react';
+import { Popover, Button, BaseControl } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { icons } from '@eightshift/frontend-libs/scripts/editor';
+import { ColorPaletteCustom } from '@eightshift/frontend-libs/scripts/components';
+
+/**
+ * Component that allows simple inline color picking while taking up not much space.
+ * 
+ * @param {object} props                                        - ColorPickerComponent options.
+ * @param {array?} props.colors                                 - List of options to display. If not set, all global manifest colors are used.
+ * @param {string} props.value                                  - Current value (color slug).
+ * @param {function} props.onChange                             - Callback that applies the changes.
+ * @param {React.Component?} props.label                        - Label to represent the control
+ * @param {string?} [props.textDomain=eightshift-frontend-libs] - Text domain to use for i18n.
+ * @param {boolean} [props.canReset=true]                       - If `true`, a clear/reset button is shown.
+ */
+export const ColorPickerComponent = ({
+	colors,
+	value,
+	onChange,
+	label,
+	textDomain = 'eightshift-frontend-libs',
```

Comment:
Is it a real use case scenario to send text-domain in `ColorPickerComponent`? I know that when you build up the `eightshift-frontend-libs` it switched depending on the project(for example when you use `eightshift-frontend-libs` in redesign web it is all searched and replaced with local textdomain - not 100% sure but I believe it is like that)

Reply:
This lives in `node_modules`, so I doubt any replacements get made, but if they do, I can very easily remove the `textDomain` and make it even simpler 😄 

At the end of the day, it's an optional parameter with a default value, so nothing will break if you forget to pass it.

Reply:
in my opinion it would be better to pass the full string from the project and not the textDomain so that the project wories about translation

Reply:
That would add probably 4-6 attributes to every component (for every string), I'm worried it would get a bit redundant and bloated if it was like this:

```jsx
<ColorPickerComponent
	label={
		<>
			{icons.color}
			<span>My color picker</span>
		</>
	}
	colors={getPaletteColors()}
	value={objData.color}
	onChange={(value) => setObjData({ color: value })}
        pickColorLabel={__('Pick a color', 'eightshift-frontend-libs')}
        buttonNoValueLabel={__('Change color', 'eightshift-frontend-libs')}
        buttonEditLabel={__('Pick a color', 'eightshift-frontend-libs')}
        resetButtonLabel={__('Reset', 'eightshift-frontend-libs')}
/>
```

Reply:
well if you have defaults on the props there is no harm in having them like this. in 99% cases you will not set any of them so ho cares if we have 2 od 20 props that you can pass. :D

Reply:
Wouldn't the defaults have the non-project text domain, though? Or does it get auto-replaced somehow? 😄

Reply:
If it goes to you project it will get replaced if not it will stay the same. Lets not waste time on this check how other core components have this and make it that way ;)

Reply:
Yes, sir!

Reply:
Looks like core is not using the second argument at all, I'll implement it like that.